File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.js
/*function solution(food) {
    var answer = '';
    let ans = [0];
    food.sort((a,b)=> b-a)
    
    food.forEach(x => {
        
        if(x>=2) {
            let foodEat = Math.floor(x/2);
            ans.push(food.indexOf(x));
            ans.unshift(food.indexOf(x));
        }
    })
    return food;
}*/

function solution(food) {
    let leftSide = [];
    
    food.forEach((item, index) => {
        if(item >= 2){
            const quotient = Math.floor(item / 2);
            
            leftSide.push(String(index).repeat(quotient)); 
        }
    })
    
    const ans = [leftSide.join(""), 0, leftSide.reverse().join("")];
    
    return ans.join("");
}

File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        StringBuilder builder = new StringBuilder();
        
        for(int i=0; i<food.length; i++) {
            
            int divNumber = food[i]/2;
             builder.append(String.valueOf(i).repeat(divNumber));
        }
        String ans = builder+"0";
        
        return ans+builder.reverse();
    }
}